<launch>
  <arg name="world_frame" default="world" doc="全局世界坐标系 ID"/>
  <arg name="base_link_frame" default="base_link" doc="机器人底盘中心坐标系 ID"/>
  
  <arg name="kinect_loop_cam_frame" default="kinect_loop_color_optical_frame" doc="Kinect_loop 彩色相机光心坐标系 ID"/>
  <arg name="kinect_circle_cam_frame" default="kinect_circle_color_optical_frame" doc="Kinect_circle 彩色相机光心坐标系 ID"/>
  
  <arg name="lidar_sensor_frame" default="lidar_link" doc="安装在底盘上的 LiDAR 传感器的参考坐标系 ID"/>
  <arg name="lidar_map_origin" default="camera_init" doc="LiDAR SLAM 的固定地图原点坐标系 ID (例如 'camera_init')"/>
  <arg name="lidar_current_pose" default="aft_mapped" doc="LiDAR 在其地图原点中的当前位姿坐标系 ID (例如 'aft_mapped')"/>

  <node pkg="tf_transformer" type="tf_transformer_node" name="tf_transformer_node" output="screen">
    <param name="world_frame_id" value="$(arg world_frame)"/>
    <param name="base_link_frame_id" value="$(arg base_link_frame)"/>
    <param name="kinect_loop_camera_optical_frame_id" value="$(arg kinect_loop_cam_frame)"/>
    <param name="kinect_circle_camera_optical_frame_id" value="$(arg kinect_circle_cam_frame)"/>
    <param name="lidar_sensor_frame_id" value="$(arg lidar_sensor_frame)"/>
    <param name="lidar_map_origin_frame_id" value="$(arg lidar_map_origin)"/>
    <param name="lidar_current_pose_frame_id" value="$(arg lidar_current_pose)"/>

    <rosparam param="world_to_lidar_map_origin">[0.0, 0.0, 0.0, 0.0, 0.0, 90.0]</rosparam>

    <rosparam param="base_link_to_kinect_loop_camera">[0.1, 0.0, 0.5, 0.0, 0.0, 0.0]</rosparam>

    <rosparam param="base_link_to_kinect_circle_camera">[0.1, 0.1, 0.4, 0.0, 0.0, 0.0]</rosparam>

    <rosparam param="base_link_to_lidar_sensor">[0.0, -0.07872, 0.357, 0.0, 0.0, 90.0]</rosparam>

    <rosparam param="lidar_current_pose_frame_to_base_link">[0.0, 0.0, 0.0, 0.0, 0.0, -90.0]</rosparam>

    </node>
</launch>